### ISSUE TITLE
why the result of detection(masks) issemantic ,  I would like to have instance 

### ISSUE THREAD

**Issue by jediofgever** [dbolya/yolact#394]
Hi,

I have trained network with my own data, using yolact_plus , my data only have one type of object,
when i run network on webcam, it does detect objects but it overlays same color onto all detected objects, meaning that it is doing semantic segmentation,
when I receive the masks as tensors, 
for example if I have 3 objects in image the masks tensor is ;
`(3,540,960,1)`
this looks correct but the content of each tensor is identical whereas each element should include only mask of one object. 
sorry if it is confusing but I needs some help at this point

**Comment by dbolya** [comment]
There may be an issue either with your training data or the model. YOLACT because of how it's designed can have an issue with overlap, but that's usually not that big of a deal. Do you get the same overlaps when using the pretrained model? And, can you check your data to see if it's proper instances?

**Comment by shivani21998** [comment]
@dbolya I'm facing the same issue. I use the pretrained model on coco to detect elephants in video. It gives the same mask color for all instances of the object instead of different colors. This is not the case for images. 

![Capture](https://user-images.githubusercontent.com/29472727/145798701-2c32c9a7-df18-40f3-b7b6-b53995eceae1.JPG)
